docs/Additional_Build_Documentation.md

```diff
@@ -0,0 +1,296 @@
+# Building Benchmark
+
+To build Benchmark we need to follow a series of simple steps:
+*   Obtain Benchmark
+*   Configure Benchmark
+*   Build Benchmark
+
+To obtain Benchmark we can simply clone the Github repository using
+```bash
+$ git clone https://github.com/google/benchmark.git
+```
+Now, Benchmark uses Google Test to run it's tests. So, if we wish to build and run
+the tests we need to include googletest, which can be done either by cloning the googletest
+repository or by asking benchmark in the configuration step to download this dependency.
+We can clone the googletest repository in benchmark using
+```bash
+$ git clone https://github.com/google/googletest.git benchmark/googletest
+```
+To configure Benchmark we use CMake as follows:
+```bash
+$ cmake -G <generator> [options] -S benchmark -B build
+```
+If we haven't cloned the googletest repository as described above we can ask benchmark to download this dependency in the above configuration step by adding `-DBENCHMARK_DOWNLOAD_DEPENDENCIES=ON`
```

Comment:
this is only if you want to build from head. if you just want to _use_ benchmark then we should document that separately.

Reply:
using has been put in a separate file now